torch/autograd/gradcheck.py

```diff
@@ -115,7 +116,9 @@ def get_analytical_jacobian(input, output):
                     if d_x is None:
                         jacobian_x[:, i].zero_()
                     else:
-                        jacobian_x[:, i] = d_x.to_dense() if d_x.is_sparse else d_x
+                        d_x_dense = d_x.to_dense() if d_x.is_sparse else d_x
+                        assert jacobian_x[:, i].numel() == d_x_dense.numel()
```

Comment:
I wanted to make sure that it didn't do any non-trivial broadcasting. For example, if `d_x` is only one element it could be broadcastable to almost any shape

Reply:
That gets to my other question; do we need to allow broadcasting at all here?  (i.e. can we just do assert jacobian_x[:, i].shape == d_x_dense.shape?

Reply:
Yes, you can get rid of the assert. It just catches some size mismatch errors a little earlier.

Reply:
That's different than what I mean.  I meant can we make the assert more restrictive, i.e. to check exact shapes rather than number of elements?

Reply:
I'm guessing this is the same as the above -- this worked with tensors because we allowed deprecated broadcasting rules?